Zend/zend_vm_def.h

```diff
@@ -9699,12 +9699,27 @@ ZEND_VM_HANDLER(167, ZEND_COPY_TMP, TMPVAR, UNUSED)
 	ZEND_VM_NEXT_OPCODE();
 }
 
-ZEND_VM_HANDLER(202, ZEND_CALLABLE_CONVERT, UNUSED, UNUSED)
+ZEND_VM_HANDLER(202, ZEND_CALLABLE_CONVERT, UNUSED, UNUSED, NUM|CACHE_SLOT)
 {
 	USE_OPLINE
 	zend_execute_data *call = EX(call);
 
-	zend_closure_from_frame(EX_VAR(opline->result.var), call);
+	if (opline->extended_value != (uint32_t)-1) {
+		int offset = (int)(uintptr_t)CACHED_PTR(opline->extended_value);
+		if (offset) {
+			offset--;
+			zend_object *closure = ((zend_object**)zend_stack_base(&EG(callable_convert_cache)))[offset];
+			ZVAL_OBJ_COPY(EX_VAR(opline->result.var), closure);
```

Comment:
Do we need that indirection?
Can't you simply store the pointer to the zend_object * in the cache?
The only reason you need the callable_convert_cache for is destruction in the non-fast-shutdown path.